### ISSUE TITLE
System.InvalidOperationException: An attempt was made to load a type with the fully qualified name of 'x' 

### ISSUE THREAD

**Issue by bugproof** [ExtendedXmlSerializer/home#374]
Hey,

I generated classes from xml schema with https://github.com/mganss/XmlSchemaClassGenerator 

The XML has attribute on the root element like `xmlns:tns="http://jpk.mf.gov.pl/wzor/2017/11/13/1113/"` but somehow ExtendedXmlSerializer can't swallow it :(. Normal XmlSerializer can but it's kinda old and would love to switch to ExtendedXmlSerializer.

This is the schema I generated class from: https://www.gov.pl/documents/2034621/2182793/Schemat_JPK_VAT%283%29_v1-1.xsd

```
Unhandled exception. System.InvalidOperationException: An attempt was made to load a type with the fully qualified name of 'http://jpk.mf.gov.pl/wzor/2017/11/13/1113/:JPK', but no type could be located with that name.
   at ExtendedXmlSerializer.ContentModel.Extensions.GetClassification(IClassification this, IFormatReader parameter, TypeInfo defaultValue)
   at ExtendedXmlSerializer.ExtensionModel.Xml.Read.Get(XmlReader parameter)
   at ExtendedXmlSerializer.ExtensionModel.Xml.Serializer.Deserialize(XmlReader reader)
   at ExtendedXmlSerializer.ExtensionModel.Xml.ExtendedXmlSerializer.Deserialize(XmlReader reader)
   at ExtendedXmlSerializer.ExtensionMethodsForSerialization.Deserialize[T](IExtendedXmlSerializer this, XmlReaderSettings settings, Stream stream)
   at ExtendedXmlSerializer.ExtensionMethodsForSerialization.Deserialize[T](IExtendedXmlSerializer this, Stream stream)
   at ConsoleApp3.Program.Main(String[] args) in C:\Users\Konrad\RiderProjects\ConsoleApp3\ConsoleApp3\Program.cs:line 18
```

Is there anyway to make it behave more like the original XmlSerializer?

```
var serializer = new ConfigurationContainer().UseAutoFormatting()
                .Create();
```

It tries to load "http://jpk.mf.gov.pl/wzor/2017/11/13/1113/:JPK" as a type but it's not a type, it's not even fully qualified type name so it should never try to load it.

This is how root node looks like:
```
<tns:JPK xsi:schemaLocation="http://jpk.mf.gov.pl/wzor/2017/11/13/1113/ https://www.gov.pl/documents/2034621/2182793/Schemat_JPK_VAT%283%29_v1-1.xsd" xmlns:tns="http://jpk.mf.gov.pl/wzor/2017/11/13/1113/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:etd="http://crd.gov.pl/xml/schematy/dziedzinowe/mf/2016/01/25/eD/DefinicjeTypy/" xmlns:kck="http://crd.gov.pl/xml/schematy/dziedzinowe/mf/2013/05/23/eD/KodyCECHKRAJOW/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:msxsl="urn:schemas-microsoft-com:xslt" xmlns:usr="urn:the-xml-files:xslt">
```

https://github.com/ExtendedXmlSerializer/home/blob/16dd56a3066896787b85a90298434315741b8108/src/ExtendedXmlSerializer/ContentModel/Extensions.cs#L23

thx in advance

**Comment by Mike-EEE** [comment]
Hi @bugproof thanks for writing in.  ExtendedXmlSerializer *should* be able to handle this, let's see. 😁

Try making use of the `InspectingType` method [as demonstrated in these tests](https://github.com/ExtendedXmlSerializer/home/blob/master/test/ExtendedXmlSerializer.Tests.ReportedIssues/Issue175Tests.cs) and see how that treats you. 🤞

**Comment by bugproof** [comment]
InspectingType gets rid of that exception but now deserialized object is empty

**Comment by Mike-EEE** [comment]
Doh!  I was afraid of that. Do note that EXS's primary use case is to [serialize/deserialize .NET object graphs and not to preserve legacy functionality](https://github.com/ExtendedXmlSerializer/home/wiki/FAQs#systemxmlserializer-vs-extendedxmlserializer).

My time these days is a bit limited for this project, unfortunately, but I will look into this and see if it's something obvious/simple to fix.  If you do have an involved scenario that cannot be easily fixed, as a worst-case scenario (preparing you here 😁), what you could do is load it in (deserialize) with the classic `XmlSerializer` and then save it (serialize) with `ExtendedXmlSerializer` (and then use that document going forward).

**Comment by bugproof** [comment]
Tried serializing as well but it doesn't serialize everything for some reason. Looks like it won't be as easy as just swapping `XmlSerializer`.

**Comment by Mike-EEE** [comment]
Wow really... so am I understanding correctly that you're getting more data emitted with ExtendedXmlSerializer, but it's not the full graph?  That should definitely not be happening as it _should_ be able to serialize pretty much anything you throw at it, especially if it is a simple graph that is serialized by the classic serializer.

Are you able to say which properties are not serializing and/or provide a precise example?  I can look further into it for you.

**Comment by bugproof** [comment]
The collection isn't serializing
```
 [System.Xml.Serialization.XmlElementAttribute("SprzedazWiersz", Namespace="http://jpk.mf.gov.pl/wzor/2017/11/13/1113/")]
 public System.Collections.ObjectModel.Collection<JPKSprzedazWiersz> SprzedazWiersz
```

**Comment by Mike-EEE** [comment]
Ah! OK @bugproof the problem we're running into is that `SprzedazWiersz` doesn't have a public setter.  Are you able to make that public in your generation?  

Also, we do support `readonly` collection getters but it's dependent on the property not having a setter at all.  I was able to get the collection to emit by doing: 

```
 public System.Collections.ObjectModel.Collection<JPKSprzedazWiersz> SprzedazWiersz { get; } =
	        new System.Collections.ObjectModel.Collection<JPKSprzedazWiersz>();
```

(And removing the initializer from the constructor).

**Comment by Mike-EEE** [comment]
I guess the other "fix" we can employ here is to add a check to the `readonly` collection getter for a private setter (in addition to no setter).

**Comment by bugproof** [comment]
Don't think it's possible to generate with public setters:
https://github.com/mganss/XmlSchemaClassGenerator/issues/9

**Comment by Mike-EEE** [comment]
Alright @bugproof after poking around at this I found another issue #200 that was similar and has a solution that involves adjustments to the extensions that are passed into the `ConfigurationContainer` constructor.  Check it out:

https://github.com/ExtendedXmlSerializer/home/blob/97f458c56de54f7993ba521484eda9d220be8730/test/ExtendedXmlSerializer.Tests.ReportedIssues/Issue374Tests.cs#L18-L47

Since we are dancing around the limitations of ExtendedXmlSerializer's legacy serialization support (trust me, it's a sore topic 😆) my preference would be to try to stay away from touching the internals as much as possible and leverage the extension model if we can.  Please let me know how that sits with you and/or any additional issues you might run into with getting this to work as desired.

**Comment by bugproof** [comment]
Should be fine to solve that serialization problem, XML in .NET is a tough topic. Probably because everyone moved on to other formats.

**Comment by Mike-EEE** [comment]
Yeah I took on the challenge of developing it, mostly to try to replicate [System.Xaml which seemed to be being deprecated at the time](https://github.com/dotnet/runtime/issues/16243). I didn't realize it also came along with the challenge of _supporting_ it, which is far far greater -- many many days now and still counting. 😆  Good learning lesson. 😁  I make this a consideration now when accepting new projects/challenges.

Anyways, we try to work with what we've got here and pray for the best results.

![](https://media2.giphy.com/media/4bpgqEqmOTwuk/source.gif)
